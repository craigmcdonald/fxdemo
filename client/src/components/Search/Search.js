import React from 'react'
import { Form, Text } from 'react-form'
import moment from 'moment'
import axios from 'axios'
import SelectBox from '../SelectBox'
import DateSelect from '../DateSelect'
import style from './Search.scss'
import PropTypes from 'prop-types'
const { array, func, object } = PropTypes

const Search = React.createClass({
  propTypes: {
    currentDate: func,
    list: array,
    results: func,
    errors: func
  },
  render () {
    const list = this.props.list
    let currentDate = this.props.currentDate()
    return (
      <div className='fx-form-outer'>
        <Form
          defaultValues={{
            base: 'GBP',
            counter: 'USD',
            date: currentDate
          }}

          onSubmit={(values) => {
            axios.get(`/exchange?date=${values.date.format('YYYY[-]MM[-]DD')}&base=${values.base}&counter=${values.counter}&amount=${values.amount}`)
            .then((response) => {
              this.props.results(response)
            })
            .catch((error) => {
              if (error.response) {
                this.props.errors(error.response)
              }
            })
          }}
        >
          {({submitForm}) => {
            return (
              <form onSubmit={submitForm}>
                <DateSelect name='date' currentDate={currentDate} />
                <SelectBox value='GBP' name='base' list={list} />
                <SelectBox value='USD' name='counter' list={list} />
                <Text
                  styleName = 'style.fx-amount'
                  field='amount'
                  placeholder='Enter amount...'
                />
                <div styleName="style.fx-submit-outer">
                  <button
                    styleName='style.fx-submit'
                    type='submit'>Convert</button>
                </div>
              </form>
            )
          }}
        </Form>
      </div>
    )
  }
})

export default Search
