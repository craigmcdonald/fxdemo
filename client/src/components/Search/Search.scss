.fx-amount {
  padding-left:10px;
  font-size: 16px;
  font-family: inherit;
  height: 33px;
  width: 238px;
  border: 1px solid #ccc;
  border-radius: 5px;
}

.fx-submit-outer {
  width:250px;
  text-align: right;
}

.fx-submit {
  margin-top: 15px;
  height: 33px;
  border-radius: 5px;
  background-color: #028E9B;
  border: 1px solid #028E9B;
  font-size: 16px;
  font-family: inherit;
  color: white;
  cursor: pointer;
}

.fx-submit:focus {
  outline:0;
}
