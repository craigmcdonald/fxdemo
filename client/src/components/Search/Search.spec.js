import React from 'react'
import ReactDOM from 'react-dom'
import Search from './Search'
import { shallow } from 'enzyme'
import moment from 'moment'
import { shallowToJson } from 'enzyme-to-json'


test('Search Snapshot Test', () => {
  const list = [['AUS','Australian Dollar']]
  const results = function(){}
  const errors = function(){}
  const currentDate = function() {
    return new Date("2017-08-09T12:53:14.878Z")
  }
  const component = shallow(
    <Search
      list={list}
      results={results}
      errors={errors}
      currentDate={currentDate}
    />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
