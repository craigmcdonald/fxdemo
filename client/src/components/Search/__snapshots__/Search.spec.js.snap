// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot Test 1`] = `
<div
  className="fx-form-outer"
>
  <Form
    defaultValues={
      Object {
        "base": "GBP",
        "counter": "USD",
        "date": 2017-08-09T12:53:14.878Z,
      }
    }
    loadState={[Function]}
    onChange={[Function]}
    onSubmit={[Function]}
    onValidationFail={[Function]}
    postSubmit={[Function]}
    preSubmit={[Function]}
    preValidate={[Function]}
    saveState={[Function]}
    validate={[Function]}
    willUnmount={[Function]}
  />
</div>
`;
